rinja/src/html.rs

```diff
@@ -1,71 +1,203 @@
-use std::fmt;
-use std::num::NonZeroU8;
+use std::{fmt, str};
 
+#[cfg(not(any(target_arch = "x86", target_arch = "x86_64")))]
 #[allow(unused)]
 pub(crate) fn write_escaped_str(mut fmt: impl fmt::Write, string: &str) -> fmt::Result {
-    let mut escaped_buf = *b"&#__;";
+    // Even though [`jetscii`] ships a generic implementation for unsupported platforms,
+    // it is not well optimized for this case. This implementation should work well enough in
+    // the meantime, until portable SIMD gets stabilized.
+
+    // Instead of testing the platform, we could test the CPU features. But given that the needed
+    // instruction set SSE 4.2 was instroduced in 2008, that it has an 99.61 % avalibility rate
```

Comment:
One more thought: wouldn't that prevent rinja to be used on embedded targets?

Reply:
I only know of Intel Galileo that uses an x86 platform for a SOC. Indeed, it does not feature any SSE instructions, but Galileo was never really a thing.

For the case that `jetscii` cannot use SSE 4.2, it implements a fallback. Unless explicitly specified as present or absent in the rustc flags, `jetscii` will query the CPU features at runtime if SSE 4.2 can be used or if the fallback has to be used. The x86-version with `jetscii`'s fallback is slower than our non-X86 version of `write_escaped_str()`, though.

We could add a build script to check if SSE 4.2 was explicitly marked as absent during the compilation, and use the generic version on x86 then, too. But I don't think the case is very likely, and I would rather wait if someone mentions that they want to use rinja on an embedded x86 system.

Reply:
Not sure if it's really that useful then if it's only for x86... Might be better to wait for portable simd to get stabilized.

Reply:
Yeah, chances are that you don't drop MBs worth of untrusted data into your templates. I guess most of the time it will be less than 16 bytes, so jetscii would not come into action anyway, but I guess different users have different use cases.

Should I copy the changed generic function into a new PR? 5% faster is not much, but I think the new code is a little more readable, too.

Reply:
More readable AND faster? Yes please.